lib/task_helpers/imports/custom_buttons.rb

```diff
@@ -18,14 +19,30 @@ def import(options)
       private
 
       class ImportArInstances
-        def self.import(obj_hash)
-          new.import(obj_hash)
+        def self.import(obj_hash, connect_dialog_by_name)
+          new.import(obj_hash, connect_dialog_by_name)
         end
 
-        def import(obj_hash)
+        def import(obj_hash, connect_dialog_by_name)
+          @connect_dialog = connect_dialog_by_name
           ActiveRecord::Base.transaction { obj_hash.each { |obj_def| create_object(*obj_def) } }
         end
 
+        def find_or_create_object(class_name, obj)
```

Comment:
In my opinion, this is to import custom buttons where they don't exist. 
In case custom buttons exist, they may use Edit task in UI to modify them.
This new method is not necessary.

Reply:
@lfu, this method is my main enhancement.
Original code raises exception when button exists.
This is unacceptable.
The import may occurs many times on the already existing buttons(in development and even in production), with or without some button changing.
In both cases the code should replace the button, optionally updating the button attributes.
The 'rhconsulting' script, which was used for porting, had this behaviour.
Your idea of using UI, seems strange to me, as all the goal of import/export scripts is to make things without UI.
Please re-think your opinion

Reply:
> The import may occurs many times on the already existing buttons(in development and even in production), with or without some button changing.

There are couple of things here.
1.  Why do you need import again if the buttons exist?
2. `name` alone is not enough to [identify a button](https://github.com/ManageIQ/manageiq/blob/abc255dd94ff8d3c1e7754b3c8a2093f8f06d309/app/models/custom_button.rb#L15).
3.  Not sure if it is a good practice to update an existing button just because it happens to have the same name. What if the buttons are owned by other users?

Reply:
1. It is needed, for example, during development, when import may be done multiple times from the same or different developers. Personally, had problems with this. Currently this will cause error,
and one need to delete manually ALL the button(s) in import directory, prior to import, or update yaml file. This is far from being convenient.
2.  You are right. Button should be looked up by class and apply_id.
This may be added to code
3. For user's support some logic exists (see check_user method). This method is untouched in PR.
It set user from export data, or 'admin'. Maybe it may be improved.
Summary:
A. Importing existed buttons needs additional efforts.
In my opinion, it is nice to have.
In case ManageIQ team has another opinion. it is not worth to implement this at all
**B. This PR includes 2 independent features: dialog support and import existing buttons.
In any way, since dialog support code is already mostly approved by you, I suggest retaining only corresponding code (for dialog support) in this PR and continue to review.**
C. Optionally, open issue about importing existing buttons.